### ISSUE TITLE
array values seem to be accepted for object-type values

### ISSUE THREAD

**Issue by ronkorving** [Wizcorp/my-type#3]
We've seen a case where a schema containing a dictionary object allowed an empty array to be assigned to that object. I guess it can be explained by the fact that to JavaScript arrays *are* objects. But I think it's obvious that `my-type` should've protected us from this.

Discovered with @tatsujinichi

**Comment by ronkorving** [comment]
Update:

Here is a code sample of what this issue causes:

```js
const schema = object({}); // empty objects only
schema.assert([]);  // passing an array, not an object

// no errors were thrown
```

and

```js
const schema = object({}); // empty objects only
schema.assert([1, 2, 3]);  // passing a non-empty array

// MyTypeError: Unknown property "0"
```

I would rather see a type error on the array, than the first property (index 0 of the array) not being allowed because of a property name violation.

**Comment by ronkorving** [comment]
Fixed by #4
